### PR TITLE
make readers.gdal streamable

### PR DESCRIPTION
**Pull request by hobu**

### CODE REVIEW

File: io/GDALReader.cpp [hunk 1]
```diff
@@ -126,81 +131,60 @@ void GDALReader::ready(PointTableRef table)
...
+    m_raster->pixelToCoord(m_col, m_row, coords);
+    double x = coords[0];
+    double y = coords[1];
+    point.setField(Dimension::Id::X, x);
+    point.setField(Dimension::Id::Y, y);
 
-    std::vector<uint8_t> band;
+    static std::vector<double> data;
 
     for (int b = 0; b < m_raster->bandCount(); ++b)
```
Comment (abellgithub) [inline io/GDALReader.cpp]: I don't think you want to do this.  You're calling m_raster->read() for each band, but it returns data for all bands.  You want the read() outside of the band loop.

File: io/GDALReader.cpp [hunk 2]
```diff
@@ -126,81 +131,60 @@ void GDALReader::ready(PointTableRef table)
...
+        m_row ++;
     }
+    m_raster->pixelToCoord(m_col, m_row, coords);
+    double x = coords[0];
+    double y = coords[1];
+    point.setField(Dimension::Id::X, x);
+    point.setField(Dimension::Id::Y, y);
 
-    std::vector<uint8_t> band;
+    static std::vector<double> data;
```
Comment (abellgithub) [inline io/GDALReader.cpp]: Making this static precludes threading.  Better just to drop the static IMO.

File: io/GDALReader.cpp [hunk 3]
```diff
@@ -81,6 +83,8 @@ void GDALReader::initialize()
     m_count = m_raster->width() * m_raster->height();
     m_bandTypes = m_raster->getPDALDimensionTypes();
     m_raster->close();
+    m_row = 0;
```
Comment (abellgithub) [inline io/GDALReader.cpp]: These belong in ready().  That allows execute() to be called more than once.  Yes, this probably never happens.

Comment (hobu) [reply io/GDALReader.cpp]: The rules of when to do which aren't so clear.

**Pull request merged**